hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java

```diff
@@ -1823,6 +1832,45 @@ public void createInSingleStartInMultiassigMode() {
         assertThat(actions).allMatch(action -> !action.getWeight().isPresent());
     }
 
+    @Test
+    @Description("Verify the that only compatible targets are part of a Rollout.")
+    void createAndStartRolloutWithTargetTypes() {
+        final int amountTargetsForRollout = 10;
+        final int amountIncompatibleTargetsForRollout = 5;
+        final String rolloutName = "rolloutTestCompatibility";
+
+        final DistributionSet testDs = testdataFactory.createDistributionSet("test-ds");
+        final TargetType targetType = testdataFactory.createTargetType("test-type", Collections.emptyList());
+
+        final List<Target> targets = testdataFactory.createTargets(amountTargetsForRollout, "testTarget-");
```

Comment:
Same as for the auto assignment test, there should also be a Target with a compatible TargetType to ensure the compatibility check is fully working, not only one without TargetType.

Reply:
will add a valid type